File: jiwon/034 시저 암호.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s, int n) {
    string answer = "";
    
    unsigned int ss[10000];
    
    for(int i=0;i<s.size();i++){
        ss[i] = s[i];
    }
    
    for(int i=0;i<s.size();i++){
        if(ss[i]>=65 && ss[i] <= 90){
            ss[i] += n;
            if(ss[i] >= 91){
                ss[i] -= 26;
            }
        } else if(ss[i]>=97 && ss[i] <= 122){
            ss[i] += n;
            if(ss[i] >= 123){
                ss[i] -= 26;
            }
        }
    }
    
    for(int i=0;i<s.size();i++){
        answer += ss[i];
    }
    
    return answer;
}

File: lesunny/034_시저암호.java
class Solution {
    public String solution(String s, int n) {
        String answer = "";
        
        char temp = ' ';
        for(int i=0; i < s.length(); i++){
            char c = s.charAt(i);
            
            if((char)65 <= c && c <= (char)90){//대문자
                temp = (char)((int)c + n);
                if(temp > 90) temp -= 26;
            }
            
            else if((char)97 <= c && c <= (char)122){//소문자
                temp = (char)((int)c + n);
                if(temp >122) temp -= 26;
            }
            
            answer += Character.toString(temp);
            temp=' ';
        }
        
        return answer;
    }
}
